### ISSUE TITLE
Client doesn't get content from a channel with resctricted saving content enabled.

### ISSUE THREAD

**Issue by umatrix81** [LonamiWebs/Telethon#3265]
If a channel has "Restrict saving content" enabled, then `event.message.message` is empty `message=''` and media is `media=MessageMediaUnsupported()`

**Comment by LightCannon** [comment]
Same problem occurs with me and I cannot find any workaround it.

**Comment by jorgeoa** [comment]
Same problem here. I hope the Telegram API allow retrieving this type of messages...

**Comment by mohinderq** [comment]
Unfortunately I do have the same problem.

**Comment by ArthurTimmermans** [comment]
Same problem ^^

**Comment by kshubham506** [comment]
Check this thread : https://github.com/LonamiWebs/Telethon/issues/3261

**Comment by mohinderq** [comment]
> Check this thread : #3261

I can confirm this is a working solution for now. Thank you! <3

**Comment by Lonami** [comment]
Likely a duplicate of #3264.

**Comment by snowman** [comment]
Simply upgrading to the latest version works for me:

```sh
$ pip install --verbose -i https://pypi.tuna.tsinghua.edu.cn/simple --upgrade telethon
$ pip show Telethon
Name: Telethon
Version: 1.25.0
Summary: Full-featured Telegram client library for Python 3
Home-page: https://github.com/LonamiWebs/Telethon
Author: Lonami Exo
Author-email: totufals@hotmail.com
License: MIT
Location: /home/w7/.local/lib/python3.10/site-packages
Requires: pyaes, rsa
Required-by:
```

**Comment by afterdelight** [comment]
yes, it was fixed in version 1.25.0!! Thanks!!!

**Comment by Robertefendi02** [comment]
Are there any updates on how to forward media from a restricted channel? I have no problem forwarding a text message, but not images/videos.

**Comment by kshubham506** [comment]
you will have to explicitly download the media and then send it, [ send the message like usual - if it throws `RESTRICTED_EXCEPTION` - download the media and call api again with file param as the input ]

**Comment by Robertefendi02** [comment]
Could you please provide some guidance or recommendation on how to download the media? Thanks

**Comment by Robertefendi02** [comment]
> you will have to explicitly download the media and then send it, [ send the message like usual - if it throws `RESTRICTED_EXCEPTION` - download the media and call api again with file param as the input ]

Could you please provide some guidance or recommendation on how to download the media? Thanks

**Comment by kshubham506** [comment]
the `.download_media` method [available in telethon](https://docs.telethon.dev/en/stable/modules/client.html#telethon.client.downloads.DownloadMethods.download_media)

**Comment by Lonami** [comment]
> you will have to explicitly download the media and then send it, [ send the message like usual - if it throws `RESTRICTED_EXCEPTION` - download the media and call api again with file param as the input ]

Would you be able to provide the full error (including exact `ERROR_TEXT`, request that caused it, and code)? This way I could add it to [`errors.csv`](https://github.com/LonamiWebs/Telethon/blob/v1/telethon_generator/data/errors.csv). Thanks!

**Comment by Robertefendi02** [comment]
> > you will have to explicitly download the media and then send it, [ send the message like usual - if it throws `RESTRICTED_EXCEPTION` - download the media and call api again with file param as the input ]
> 
> Would you be able to provide the full error (including exact `ERROR_TEXT`, request that caused it, and code)? This way I could add it to [`errors.csv`](https://github.com/LonamiWebs/Telethon/blob/v1/telethon_generator/data/errors.csv). Thanks!

I used the above recommendation ".download_media method [available in telethon]" in my own code and it worked. After using "try and except e:" I got this error "(caused by SendMediaRequest)". Is there a way to determine if this message was sent from a restricted channel? Is there any type of tag available in event. message which show the message was send from restricted channel?
